src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import static core.basesyntax.Figure.DEFAULT_COLOR;
+import static core.basesyntax.Figure.DEFAULT_RADIUS;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_TYPES = 5;
+    private static final double DIMENSION_RANGE = 10.0;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier colorSupplier) {
```

Comment:
let's make FigureSupplier more isolated, so initialise these parameters during the declaration
```
private final Random random = new Random();
private final ColorSupplier colorSupplier = new ColorSupplier();
```